dandi/dandiapi.py

```diff
@@ -436,19 +436,21 @@ def publish(self) -> "RemoteDandiset":
         `Version`.
         """
         lgr.debug("Waiting for Dandiset %s to complete validation ...", self.identifier)
-        while True:
+        start = time()
+        # Wait 20s for celery tasks to finish
+        while time() - start < 20:
```

Comment:
why is sleeping in this api file. this should only depend on whether the dataset is valid or not. the process of sleeping should be left up to any client depending on the tasks it does.

Reply:
shouldn't any wait be in the test related to publish?

Reply:
This wait is currently only used before publishing a dandiset, so it lives in the publish function. We could move it out into a separate `wait_until_valid` function @jwodder ?

Reply:
@dchiquito what values "state" could be -- only "Valid" and "Invalid" (docs in https://api.dandiarchive.org/swagger/ tell nothing about any of expected values)?  What is the "life-cycle" for/between those two states?

Reply:
but why wait and for how long (i mean practically in a real dataset scenario) ? a dataset could have multiple 5TB objects whose checksum computation could take hours. that's why in the context of a test, one should know what a potential time window is and wait for that time in the test itself, not here in the publish function.

Reply:
I agree that splitting the waiting into a separate `wait_until_valid()` method  (perhaps with a `min_time` parameter) makes sense.